### ISSUE TITLE
`:has_one` field doesn't respect disabled

### ISSUE THREAD

**Issue by krystof-k** [avo-hq/avo#2614]
### Describe the bug

When using `:has_one` association field, it doesn't respect `disabled: true` when the association is already present (you can remove it). When there is no association, no create association button is visible.

### Steps to Reproduce
1. Add `:has_one` association with `disabled: true`
   ```ruby
      field :user,
        as: :has_one,
        disabled: true
2. Go to the record and try to remove the association

### Expected behavior & Actual behavior

It should not allow to remove/delete/edit the association.

### Models and resource files
<!-- If it helps with replicating the issue, please add models and resources configuration. -->

### System configuration
**Avo version**: 3.5.3

**Rails version**: 6.1.7.7

**Ruby version**: 3.1.4

**License type**:
<!-- (Mark [x] inside the brackets) -->

 - [x] Community
 - [ ] Pro
 - [ ] Advanced

**Are you using Avo monkey patches, overriding views or view components?**
<!-- (Mark [x] inside the brackets) -->

 - [ ] Yes. If so, please post code samples.
 - [x] No

### Screenshots or screen recordings
<!-- If applicable, add screenshots or recordings to help explain your problem. -->

### Additional context
<!-- Add any other context about the problem here. -->

#### Impact
<!--
  How much are you impacted by this issue?
  Please set a level of Impact and Urgency
  (Mark [x] inside the brackets)
-->

 - [ ] High impact (It makes my app un-usable.)
 - [x] Medium impact (I'm annoyed, but I'll live.)
 - [ ] Low impact (It's really a tiny thing that I could live with.)

#### Urgency

 - [ ] High urgency (I can't continue development without it.)
 - [ ] Medium urgency (I found a workaround, but I'd love to have it fixed.)
 - [x] Low urgency (It can wait. I just wanted you to know about it.)

**Comment by krystof-k** [comment]
`:has_many` behaves probably the same.

**Comment by Paul-Bob** [comment]
@krystof-k `disabled` and `readonly` are form input arguments. The mentioned association fields are not inputs on a form so that options do not affect them. If you want to have some granular control over the permitted actions on that fields the best approach is to use authorization https://docs.avohq.io/3.0/authorization.html#associations

**Comment by krystof-k** [comment]
OK, but it does affect the buttons when there is no association, so I'm a little confused.

`disabled: false`:

![image](https://github.com/avo-hq/avo/assets/11633481/19022164-4b35-45eb-ae0b-f29af0600073)

`disabled: true`:

![image](https://github.com/avo-hq/avo/assets/11633481/dacf53e0-8623-4a11-b1a4-f69258f6460f)

I think it would be reasonable to `disabled` work for all fields correctly.

I know it is possible to use the authorization and I get the @adrianthedev's argument in #2552, but I think this is a little confusing and I understand the authorization as something which controls _who can do what_, whereas this is related to _what is possible where_.

**Comment by Paul-Bob** [comment]
Good point on the `disabled` option, should be coherent and disable all buttons or disable none.

> I understand the authorization as something which controls who can do what, whereas this is related to what is possible where.

Understand it but I think that both can be achieved in one single point, on a policy file... Having multiple places to restrict actions is confusing. You can use authorization to control by role (who can do what) or just `true` / `false` (what is possible where)

**Comment by krystof-k** [comment]
Yeah, sure it can, that is your design decision. In that case, what about raising an error when `disabled: true` is used for an association field? However for `:has_one`, it works fine with disabling the select box, so I'm not sure, complicated…

**Comment by adrianthedev** [comment]
I understand the confusion with `disabled`, but the truth is that the `has_one` has more than just one state. A user can attach/detach/create records on it, so that's why we chose to control that using policies as we control the rest of the associations.

> Yeah, sure it can, that is your design decision. In that case, what about raising an error when `disabled: true` is used for an association field? However for `:has_one`, it works fine with disabling the select box, so I'm not sure, complicated…

We'll take a PR for that.

**Comment by Paul-Bob** [comment]
We decided to make it coherent by removing the `disabled` option on associations fields. I.E `disabled: true` will stop working even when there is no attached record. Associations controls should be entirely controlled by policies and custom controls. Thanks for reporting this and for all the details @krystof-k

**Comment by Paul-Bob** [comment]
```ruby
field :admin, as: :has_one
field :team_members, as: :has_many, through: :memberships
```
![Image](https://github.com/user-attachments/assets/cc4ebd70-6779-4f5e-813e-73d585e2304b)

---

```ruby
field :admin, as: :has_one, disabled: true
field :team_members, as: :has_many, through: :memberships, disabled: true
```

![Image](https://github.com/user-attachments/assets/5274d59d-4476-41e9-bb8d-de654957bf61)


---

### Approach
- Remove the `disabled` and `read_only` options from association fields, ensuring they no longer have any functional impact.
  - References:
    - [`show_component.html.erb#L8`](https://github.com/avo-hq/avo/blob/main/app/components/avo/fields/has_one_field/show_component.html.erb#L8)
    - [`show_component.html.erb#L20`](https://github.com/avo-hq/avo/blob/main/app/components/avo/fields/has_one_field/show_component.html.erb#L20)
- Document this as a breaking change, noting that this decision was made to maintain consistency across association field types. The breaking change documentation should advise users to search for instances of the `disabled: true` option and ensure that any relevant use cases are handled by appropriate policy settings.
